File: Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/model/invocation/ConvertInvocation.java
package org.cqframework.cql.cql2elm.model.invocation;

import java.util.Collections;
import java.util.List;
import org.hl7.elm.r1.Convert;
import org.hl7.elm.r1.Expression;

public class ConvertInvocation extends OperatorExpressionInvocation<Convert> {
    public ConvertInvocation(Convert expression) {
        super(expression);
    }

    @Override
    public List<Expression> getOperands() {
        return Collections.singletonList(expression.getOperand());
    }

    @Override
    public void setOperands(List<Expression> operands) {
        expression.setOperand(requireSingleton(operands));
    }
}

File: Src/java/cql-to-elm/src/main/kotlin/org/cqframework/cql/cql2elm/model/invocation/ConvertInvocation.kt
package org.cqframework.cql.cql2elm.model.invocation

import org.hl7.elm.r1.Convert
import org.hl7.elm.r1.Expression

class ConvertInvocation(expression: Convert) : OperatorExpressionInvocation<Convert>(expression) {
    override var operands: List<Expression>
        get() = listOf(expression.operand!!)
        set(operands) {
            require(operands.size == 1) { "Unary operator expected." }
            expression.operand = operands[0]
        }
}
